File: 0_数组/旋转数组/1752. 检查数组是否经排序和轮转得到.ts
// 小技巧，搜索前对字符串变换
// 1764. 通过连接另一个数组的子数组得到一个数组
// function check(nums: number[]): boolean {
//   const str1 = nums.map(num => `#${num}#`).join('')
//   const str2 = nums
//     .slice()
//     .sort((a, b) => a - b)
//     .map(num => `#${num}#`)
//     .join('')

//   return isFlipedString(str1, str2)
// }

// 至多存在一个下降点
function sum2(nums: number[]): boolean {
  let down = 0

  for (let i = 0; i < nums.length; i++) {
    if (nums[i] > nums[(i + 1) % nums.length]) down++
  }

  return down <= 1
}

console.log(sum2([3, 4, 5, 1, 2]))

File: 0_数组/旋转数组/1752. 检查数组是否经排序和轮转得到.py
from typing import List


class Solution:
    def check(self, nums: List[int]) -> bool:
        n = len(nums)
        down = 0
        for i in range(n):
            if nums[i] > nums[(i + 1) % n]:
                down += 1
            if down > 1:
                return False
        return True
